src/libstd/collections/hash/map.rs

```diff
@@ -2445,13 +2445,18 @@ impl RandomState {
         // iteration order allows a form of DOS attack. To counter that we
         // increment one of the seeds on every RandomState creation, giving
         // every corresponding HashMap a different iteration order.
-        thread_local!(static KEYS: Cell<(u64, u64)> = {
-            Cell::new(sys::hashmap_random_keys())
+        thread_local!(static KEYS: Cell<[u64; 2]> = {
+            let mut buf = [0u8; 16];
+            // we consciously ignore potential errors here
+            let _ = getrandom::getrandom(&mut buf);
```

Comment:
All of the `hashmap_random_keys` implementations would `panic` on error, is there a reason to switch to ignoring them?

Reply:
Only those which retrieve OS entropy, others like Redox or WASM use a fixed constant. Considering that OS RNG failure should be extremely rare on supported platforms, I think it's reasonable to ignore potential errors here. Also it will allow to "[support](https://github.com/rust-lang/rust/issues/62079#issuecomment-504769548)" new targets automatically without CI breakage.

Reply:
There is precedent for this — the WASM and Redox implementations (deleted by this PR) simply return a constant. That said, this changes things in that all platforms now may lose hash-map randomisation if `getrandom` fails (though as @newpavlov says, such failures should be extremely rare).

I personally don't want to make the call on this one. (Alternatively one could choose a platform-specific approach.)

Reply:
How about panicking if the error isn't `Error::UNAVAILABLE`?